### DISCUSSION TITLE
Automatic Data Plane Registration on Start-up

### DISCUSSION THREAD

**Discussion by ghost** [eclipse-edc/Connector#2188] [Ideas]
## Setup
We deployed some EDC instances using different methods like docker-compose or helm charts. In such a setup, we separated into one data plane and one control plane with a postgreSQL database.

## The Problem
Our goal is to create a setup that can be used with a simple `docker-compose up` which creates a working stack. The `docker-compose.yml` looks like this:
```
version: '3.8'

services:
  control-plane:
    ...
  data-plane:
    ...
  postgres-control-plane:
    ...
```

After starting this stack, we always have to register the consumer data plane manually by calling the control plane endpoint https://control-plane.example.com/api/v1/dataplane/instances with the following payload:
```
{
    "edctype": "dataspaceconnector:dataplaneinstance",
    "id": "http-data-plane",
    "url": "CONSUMER-DATA-PLANE-CONTROL/transfer",
    "allowedSourceTypes": [
        "HttpData", "HttpProvision"
    ],
    "allowedDestTypes": [
        "HttpData", "HttpProxy"
    ],
    "properties": {
        "publicApiUrl": "CONSUMER-DATA-PLANE-PUBLIC"
    }
}
```

This manual step is unfortunately somewhat laborious. To solve this problem, we created a simple init script to automate the process and integrate it into the docker-compose setup by adding another container that is only used for this control plane initialization:

```
version: '3.8'

services:
  control-plane:
    ...
  data-plane:
    ...
  postgres-control-plane:
    ...
  init-data-plane-control-plane:
    ...
```
 
## Possible Solution
To prevent this manual step, it would be great to find a possibility to automatically register the data plane at the control plane. Perhaps by adding another attribute(s) in the configuration file with the url. On startup, the above mentioned endpoint could be called automatically and the payload could be filled with the configuration attribute(s). This would eliminate the manual call / the initialization container.

Would be great if we can discuss this proposal or think about other solutions!

**Comment by ndr-brt** [comment]
Another solution could be to provide an extension to the data-plane that will call the data-plane-resolver endponit and register itself at the startup.
I'm not sure that this could be something to integrate in the upstream, but we'd happy to review an eventual proposal.

**Reply by ndr-brt** [reply]
I presented the same question in a more succinct way: https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/discussions/2198
